compiler/rustc_hir_analysis/src/check/wfcheck.rs

```diff
@@ -1880,18 +1880,20 @@ pub fn check_variances_for_type_defn<'tcx>(
                 hir_param.span,
                 "hir generics and ty generics in different order",
             );
-            continue;
-        }
-
-        match hir_param.name {
-            hir::ParamName::Error => {}
-            _ => {
-                let has_explicit_bounds = explicitly_bounded_params.contains(&parameter);
-                res = Err(report_bivariance(tcx, hir_param, has_explicit_bounds, item));
+        } else {
+            match hir_param.name {
+                hir::ParamName::Error => {}
+                _ => {
+                    let has_explicit_bounds = explicitly_bounded_params.contains(&parameter);
+                    report_bivariance(tcx, hir_param, has_explicit_bounds, item);
+                }
             }
-        }
+        };
+        let mut v = variances.to_vec();
+        v[index] = ty::Invariant;
+        variances = tcx.arena.alloc_slice(&v);
     }
-    res
+    Ok(variances)
```

Comment:
Does this function ever return `Err`?

Unless I'm being particularly unobservant, maybe let's rename this fn to `check_variances_for_type_defn_and_set_unconstrained_to_invariant`

(long name but there's only a few callsites -- happy to workshop the name lol

Reply:
edit: nvm I guess lol

Reply:
I could return a vec of erroneous indices instead of doing the modification and the checks. Seems cleaner to separate these things